hooks/useFuturesData.ts

```diff
@@ -0,0 +1,268 @@
+import React, { useState, useMemo, useEffect } from 'react';
+import { useRecoilState, useRecoilValue } from 'recoil';
+import { useRouter } from 'next/router';
+import useSynthetixQueries from '@synthetixio/queries';
+import Wei, { wei } from '@synthetixio/wei';
+import { ethers } from 'ethers';
+
+import { gasSpeedState, walletAddressState } from 'store/wallet';
+import useExchangeRatesQuery from 'queries/rates/useExchangeRatesQuery';
+import Connector from 'containers/Connector';
+import { getMarketKey } from 'utils/futures';
+import { Synths } from 'constants/currency';
+import useGetFuturesMarketLimit from 'queries/futures/useGetFuturesMarketLimit';
+import {
+	currentMarketState,
+	feeCostState,
+	leverageSideState,
+	leverageState,
+	leverageValueCommitedState,
+	marketInfoState,
+	maxLeverageState,
+	orderTypeState,
+	positionState,
+	sizeDeltaState,
+	tradeSizeState,
+	tradeSizeSUSDState,
+} from 'store/futures';
+import { newGetExchangeRatesForCurrencies } from 'utils/currencies';
+import { PositionSide } from 'sections/futures/types';
+import { getFuturesMarketContract } from 'queries/futures/utils';
+import { zeroBN } from 'utils/formatters/number';
+import { KWENTA_TRACKING_CODE } from 'queries/futures/constants';
+import TransactionNotifier from 'containers/TransactionNotifier';
+import RefetchContext from 'contexts/RefetchContext';
+
+const DEFAULT_MAX_LEVERAGE = wei(10);
+
+const useFuturesData = () => {
+	const walletAddress = useRecoilValue(walletAddressState);
+	const exchangeRatesQuery = useExchangeRatesQuery();
+	const router = useRouter();
+	const { synthetixjs, network } = Connector.useContainer();
+	const { useSynthetixTxn, useEthGasPriceQuery } = useSynthetixQueries();
+	const { monitorTransaction } = TransactionNotifier.useContainer();
+	const { handleRefetch } = React.useContext(RefetchContext);
```

Comment:
A nice little helper I've found in the past is to export a hook in the context like `useRefetchContext` just to wrap in it one place instead of calling through useContext on each use.

`const useRefetchContext = () => React.useContext(RefetchContext)`

Reply:
True. Thanks @avclarke. Will add the hook for this.